daemon/inertiad/crypto/key_derivation.go

```diff
@@ -0,0 +1,34 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"crypto/sha1"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+const (
+	// KeyDerivationSaltLength is the length of the salt in bytes
+	KeyDerivationSaltLength = 8
+	// KeyDerivationKeyLength is the length of the key derived in bytes
+	KeyDerivationKeyLength  = 32
+	keyDerivationIterations = 10000
+)
+
+// GenerateSalt returns a random hex encoded salt for KD algorithm
+func GenerateSalt() []byte {
+	salt := make([]byte, KeyDerivationSaltLength)
+	rand.Read(salt)
+	return salt
+}
+
+// DeriveKey derives an AES encryption key based on salt + user's
+// password using PBKDF2 with HMAC-SHA1
+func DeriveKey(password string, salt []byte) []byte {
+	return pbkdf2.Key(
+		[]byte(password),
+		salt,
+		keyDerivationIterations,
+		KeyDerivationKeyLength,
+		sha1.New,
```

Comment:
_Please_ can we not use SHA1.

Reply:
Yeah we can use SHA-256 if we want.

But FYI there's no weakness in SHA-1 when it's used for HMAC:

https://stackoverflow.com/a/4944822

Whats the issue?

Reply:
should be fixed